packages/core/src/amazonq/lsp/lspClient.ts

```diff
@@ -326,42 +357,23 @@ export async function activate(extensionContext: ExtensionContext) {
             if (savedDocument && editor && editor.document.uri.fsPath !== savedDocument.fsPath) {
                 void LspClient.instance.updateIndex([savedDocument.fsPath], 'update')
             }
+            // user created a new empty file using File -> New File
+            // these events will not be captured by vscode.workspace.onDidCreateFiles
+            // because it was created by File Explorer(Win) or Finder(MacOS)
+            // TODO: consider using a high performance fs watcher
```

Comment:
? fs watchers have been a source of performance issues in the past. Please do not add more of them without careful consideration. (We already spawn tons of them.)